### ISSUE TITLE
memory leak while convert Pixmap's colorspace

### ISSUE THREAD

**Issue by heardawei** [pymupdf/PyMuPDF#4125]
### Description of the bug

call `Pixmap` 's `__init__(Colorspace, Pixmap)` override will cause a memory leak when the source pixmap's colorspace don't equal fitz.csRGB and the target colorspace is fitz.csRGB.

### How to reproduce the bug

pdf: [demo.pdf](https://github.com/user-attachments/files/18059176/demo.pdf)

just run code to reproduce:
```
import fitz
import gc
import psutil


def main():
    """
    Check for memory leaks.
    """
    process = psutil.Process()

    def get_rss():
        return round(process.memory_info().rss / (1024 * 1024), 2)

    stat = get_rss()
    demo = "demo.pdf"

    for i in range(10):
        with fitz.open(demo) as doc:
            for one_page in doc:
                for info in one_page.get_images(full=True):
                    pix = fitz.Pixmap(doc, info[0])
                    if pix.colorspace != fitz.csRGB:
                        pix1 = fitz.Pixmap(fitz.csRGB, pix)
                        del pix1
                    del pix
                break
        fitz.TOOLS.store_shrink(100)
        fitz.TOOLS.glyph_cache_empty()
        gc.collect()
        stat1 = get_rss()
        print(
            f"init RSS: {stat:.3f}MB -> current RSS: {stat1:.3f}MB, leak: {stat1-stat:.3f}MB"
        )

    stat1 = get_rss()
    print(
        f"init RSS: {stat:.3f}MB -> final RSS: {stat1:.3f}MB, leak: {stat1-stat:.3f}MB"
    )


if __name__ == "__main__":
    main()
```

### PyMuPDF version

1.24.9

### Operating system

Linux

### Python version

3.12

**Comment by julian-smith-artifex-com** [comment]
Many thanks for this, i've confirmed it's a genuine leak.

It's a problem in MuPDF, and i have a fix in my tree.

Hopefully we'll be able to include the fix in the next PyMuPDF release, This isn't certain because of it being in MuPDF rather than PyMuPDF, but we'll do our best to get it released sooner rather than later.

**Comment by heardawei** [comment]
can you help me fix it for temporary？thanks very much

**Comment by julian-smith-artifex-com** [comment]
> can you help me fix it for temporary？thanks very much

It ended up being easier to make a new fixed release than to create usable docs on how to patch MuPDF and build PyMuPDF from source.

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.25.1.

**Comment by py100** [comment]
> Fixed in PyMuPDF-1.25.1.

How can I fix this bug in 1.24.10，I have some trouble updating PyMuPDF>=1.25.1

**Comment by julian-smith-artifex-com** [comment]
There is no supported mechanism to fix this in 1.24.10, or indeed fix any old releases.

You need to upgrade to the latest release, PyMuPDF-1.26.0, which was released yesterday. If you have problems doing this, please create a new Github issue and we'll do our best to help.
